File: ios/NativeEncryption.m
//
//  NativEncryption.m
//  Copyright © 2019 Facebook. All rights reserved.
//

#import <Foundation/Foundation.h>


#import <React/RCTBridgeModule.h>
#import <React/RCTEventDispatcher.h>

@interface RCT_EXTERN_MODULE(NativeEncryption, NSObject)

RCT_EXTERN_METHOD(encryptData:(NSString *)key iv:(NSString *)iv str:(NSString *)str callback:(RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(decryptData:(NSString *)key iv:(NSString *)iv str:(NSString *)str callback:(RCTResponseSenderBlock)callback)

@end

File: ios/NativeEncryption.swift

//  NativEncryption.swift
//  Copyright © 2019 Facebook. All rights reserved.
//
import Foundation

@objc(NativeEncryption)
class NativeEncryption: NSObject {
  
  @objc func encryptData(_ key : String,iv : String,str : String,callback: RCTResponseSenderBlock) -> Void{
    callback([NSNull(),CryptLib().encryptPlainText(with: str, key: key, iv: iv) ?? ""])
  }
  
  @objc func decryptData(_ key : String,iv : String,str : String,callback: RCTResponseSenderBlock) -> Void {
    callback([NSNull(),CryptLib().decryptCipherText(with: str, key: key, iv: iv) ?? ""])
  }
  
}
